libraries/chain/block_log.cpp

```diff
@@ -496,19 +496,21 @@ namespace eosio { namespace chain {
 
       /// Would remove pre-existing block log and index, never write blocks into disk.
       struct empty_block_log final : block_log_impl {
+         uint32_t first_block_number = std::numeric_limits<uint32_t>::max();
```

Comment:
Why not initialize to `1`?

Reply:
Because `first_block_num()` is suppose to report the first available block in the block log. The `net_plugin` for example will report that to peers so they can sync from that block number. So an empty block log should only report what it has which is just the current head or nothing.